cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java

```diff
@@ -98,7 +100,9 @@
   DistributedProgramRuntimeService(ProgramRunnerFactory programRunnerFactory, TwillRunner twillRunner, Store store,
                                    MetricsCollectionService metricsCollectionService,
                                    Configuration hConf, CConfiguration cConf,
-                                   ArtifactRepository artifactRepository, Impersonator impersonator,
+                                   @Named(AppFabricServiceRuntimeModule.BASE_ARTIFACT_REPO)
+                                     ArtifactRepository artifactRepository,
```

Comment:
Let's add a comment on why bypassing authorization is okay here, and in other places.